### PR TITLE
Custom component in Fold fix

### PR DESCRIPTION
**Pull request by VeraZab**
as a fix for: #386 #385

### CODE REVIEW

**Comment by VeraZab** [comment]
@nicolaskruchten I can't really think of anything that would break if we add this which would allow for external apps not to have to put in anything special if they want a custom component to appear in a fold. Can you?

**Comment by nicolaskruchten** [comment]
This is basically similar to the `Section` pattern right, where non-`attr`-containing components count as visible by default?

**Comment by VeraZab** [comment]
hmm, well, here.. even if all children are non visible.. we still want the parent to appear, like in the case with the color bar here
screen shot 2018-03-07 at 2 27 24 pm

and nesting of folds is important too whereas in Sections not really sure how that looks like. We don't want the child folds to appear if they don't contain anything:
screen shot 2018-03-07 at 2 22 59 pm

File: src/components/containers/Fold.js [hunk 1]
```diff
@@ -26,18 +26,30 @@ class Fold extends Component {
     this.foldVisible = false;
 
     React.Children.forEach(nextProps.children, child => {
-      if (child.props.attr) {
-        let plotProps;
-        if (child.type.supplyPlotProps) {
-          plotProps = child.type.supplyPlotProps(child.props, nextContext);
-          if (child.type.modifyPlotProps) {
-            child.type.modifyPlotProps(child.props, nextContext, plotProps);
+      if (child) {
```
Comment (nicolaskruchten) [inline src/components/containers/Fold.js]: could these two `if`s be merged?

File: src/components/containers/Fold.js [hunk 2]
```diff
@@ -26,18 +26,30 @@ class Fold extends Component {
...
+            plotProps = child.type.supplyPlotProps(child.props, nextContext);
+            if (child.type.modifyPlotProps) {
+              child.type.modifyPlotProps(child.props, nextContext, plotProps);
+            }
+          } else {
+            plotProps = unpackPlotProps(child.props, nextContext);
+          }
+
+          if (plotProps.isVisible) {
+            this.foldVisible = true;
```
Comment (nicolaskruchten) [inline src/components/containers/Fold.js]: where is the case handled where the child has no `attr` ?

Comment (nicolaskruchten) [reply src/components/containers/Fold.js]: ah I see, below.

Comment (nicolaskruchten) [reply src/components/containers/Fold.js]: ah, no, I don't see...

**Comment by nicolaskruchten** [comment]
💃

**Comment by VeraZab** [comment]
ok @bpostlethwaite want to try master with this fix?
should have fixed your 2 issues above!

**Comment by bpostlethwaite** [comment]
Thanks @VeraZab will do!

**Pull request merged**